Core/include/Acts/TrackFinder/CombinatorialKalmanFilter.hpp

```diff
@@ -0,0 +1,1255 @@
+      state.stepping.navDir = backward;
+      // Set accumulatd path to zero before targeting surface
+      state.stepping.pathAccumulated = 0.;
+      // Not sure if the following line helps anything
+      state.options.direction = backward;
+
+      return Result<void>::success();
+    }
+
+    /// Pointer to a logger that is owned by the parent,
+    /// CombinatorialKalmanFilter
+    const Logger* m_logger;
+
+    /// Getter for the logger, to support logging macros
+    const Logger& logger() const { return *m_logger; }
+
+    /// The CombinatorialKalmanFilter updater
+    updater_t m_updater;
+
+    /// The CombinatorialKalmanFilter smoother
+    smoother_t m_smoother;
+
+    /// The source link selector
+    source_link_selector_t m_sourcelinkSelector;
+
+    /// The branch propagation stopper
+    branch_stopper_t m_branchStopper;
+
+    /// The Measuremetn calibrator
+    calibrator_t m_calibrator;
+
+    /// The Surface beeing
+    SurfaceReached targetReached;
+  };
+
+  template <typename source_link_t, typename parameters_t>
+  class Aborter {
+   public:
+    /// Broadcast the result_type
+    using action_type = Actor<source_link_t, parameters_t>;
+
+    template <typename propagator_state_t, typename stepper_t,
+              typename result_t>
+    bool operator()(propagator_state_t& /*state*/, const stepper_t& /*stepper*/,
+                    const result_t& result) const {
+      if (!result.result.ok()) {
+        return true;
+      }
+      return false;
+    }
+  };
+
+ public:
+  /// Fit implementation of the foward filter, calls the
+  /// the forward filter and backward smoother
+  ///
+  /// @tparam source_link_t Source link type identifying uncalibrated input
+  /// measurements.
+  /// @tparam start_parameters_t Type of the initial parameters
+  /// @tparam comb_kalman_filter_options_t Type of the CombinatorialKalmanFilter
+  /// options
+  /// @tparam parameters_t Type of parameters used for local parameters
+  ///
+  /// @param sourcelinks The fittable uncalibrated measurements
+  /// @param sParameters The initial track parameters
+  /// @param tfOptions CombinatorialKalmanFilterOptions steering the track
+  /// finding
+  /// @note The input measurements are given in the form of @c SourceLinks.
+  /// It's
+  /// @c calibrator_t's job to turn them into calibrated measurements used in
+  /// the track finding.
+  ///
+  /// @return the output as an output track
+  template <typename source_link_t, typename start_parameters_t,
+            typename comb_kalman_filter_options_t,
+            typename parameters_t = BoundParameters,
+            typename result_t =
+                Result<CombinatorialKalmanFilterResult<source_link_t>>>
+  auto findTracks(const std::vector<source_link_t>& sourcelinks,
+                  const start_parameters_t& sParameters,
+                  const comb_kalman_filter_options_t& tfOptions) const
+      -> std::enable_if_t<!isDirectNavigator, result_t> {
```

Comment:
Never seen this way of SFINAE, before. Sweet!

Reply:
If you removed `DirectNavigator` from the CKF, this SFINAE overload can probably be the only one, and the one where `isDirectNavigator == true` can be removed, right?

Reply:
Exactly. All of these have been removed (if you look at the updated code).

Reply:
Ah right, ok. Didn't see that then.